File: ios/Runner/Handler/TimerChannelHandler.swift
//
//  TimerChannelHandler.swift
//  Runner
//
//  Created by asimsharf on 15/11/2024.
//


import Flutter
import ActivityKit

class TimerChannelHandler {
    private static let channelName = "com.example.dynamicIsland/timerService"

    static func setUpChannel(with controller: FlutterViewController) {
        let channel = FlutterMethodChannel(name: channelName, binaryMessenger: controller.binaryMessenger)
        
        channel.setMethodCallHandler { call, result in
            switch call.method {
            case "startTimer":
                if let args = call.arguments as? [String: Any],
                   let duration = args["duration"] as? Double {
                    TimerActivityService.shared.startCountdownTimer(durationInSeconds: duration)
                    result("Countdown timer started")
                }
            case "stopTimer":
                TimerActivityService.shared.stopAllActivities()
                result("All timers stopped")
            default:
                result(FlutterMethodNotImplemented)
            }
        }
    }
}

File: android/app/src/main/kotlin/com/example/liveactivity/handlers/TimerChannelHandler.kt
package com.example.liveactivity.handlers

import android.content.Context
import io.flutter.embedding.engine.dart.DartExecutor
import io.flutter.plugin.common.MethodChannel
import com.example.liveactivity.helpers.TimerServiceHelper

class TimerChannelHandler(private val context: Context) {
    companion object {
        const val CHANNEL_NAME = "com.example.liveactivity/timerService"
    }

    fun setUpChannel(dartExecutor: DartExecutor) {
        val  channel = MethodChannel(dartExecutor.binaryMessenger, CHANNEL_NAME)
        channel.setMethodCallHandler { call, result ->
            when (call.method) {
                "startTimer" -> {
                    val duration = call.argument<Int>("duration") ?: 0
                    TimerServiceHelper.startTimerService(context, duration)
                    result.success("Timer started")
                }
                "stopTimer" -> {
                    TimerServiceHelper.stopTimerService(context)
                    result.success("Timer stopped")
                }
                else -> result.notImplemented()
            }
        }

    }
}